src/components/create-filter-form.js

```diff
@@ -91,30 +95,82 @@ const CreateFilterDialog = ({
     customTextValidationBtn,
     customTextCancelBtn,
 }) => {
-    const [disableBtnSave, setDisableBtnSave] = useState(true);
     const [newNameList, setNewListName] = useState('');
     const [newListType, setNewListType] = useState(FilterType.SCRIPT);
     const [filterPrivacy, setFilterPrivacy] = React.useState('private');
     const [createFilterErr, setCreateFilterErr] = React.useState('');
     const activeDirectory = useSelector((state) => state.activeDirectory);
 
+    const [filterInvalid, setFilterInvalid] = useState(false);
+    const [loadingCheckFilterName, setLoadingCheckFilterName] =
+        React.useState(false);
+    const [filterNameChecked, setFilterNameChecked] = React.useState(false);
+
+    const classes = useStyles();
+    const intl = useIntl();
+    const timer = React.useRef();
+
     /**
      * on change input popup check if name already exist
      * @param name
      */
-    const onChangeInputName = (name) => {
-        if (name.length === 0) {
-            setDisableBtnSave(true);
+    const updateFilterFormState = (name) => {
+        if (name !== '') {
+            //If the name is not only white spaces
+            if (name.replace(/ /g, '') !== '') {
+                elementExists(activeDirectory, name, ElementType.FILTER)
+                    .then((data) => {
+                        setFilterFormState(
+                            data
+                                ? intl.formatMessage({
+                                      id: 'nameAlreadyUsed',
+                                  })
+                                : '',
+                            !data
+                        );
+                    })
+                    .catch((error) => {
+                        setCreateFilterErr(
+                            intl.formatMessage({
+                                id: 'nameValidityCheckErrorMsg',
+                            }) + error
+                        );
+                    });
+            } else {
+                setFilterFormState(
+                    intl.formatMessage({ id: 'nameEmpty' }),
+                    false
+                );
+            }
         } else {
-            setNewListName(name);
-            setDisableBtnSave(false);
+            setFilterFormState('', false);
         }
+        setLoadingCheckFilterName(false);
```

Comment:
Maybe it should be in a .finally for the elementExists call and in some else as well, because here it's a quick request but with this actual code you could setLoadingCheckFilterName(false); before request done.